confignode/src/main/java/org/apache/iotdb/confignode/consensus/request/write/CreateSchemaPartitionReq.java

```diff
@@ -18,74 +18,71 @@
+
 import java.io.DataOutputStream;
 import java.io.IOException;
 import java.nio.ByteBuffer;
 import java.util.HashMap;
 import java.util.Map;
-import java.util.Map.Entry;
 import java.util.Objects;
 
 /** Create SchemaPartition by assignedSchemaPartition */
 public class CreateSchemaPartitionReq extends ConfigRequest {
 
-  private Map<String, Map<TSeriesPartitionSlot, TRegionReplicaSet>> assignedSchemaPartition;
+  // TODO: Replace this field whit new SchemaPartition
+  private Map<String, SchemaPartitionTable> assignedSchemaPartition;
 
   public CreateSchemaPartitionReq() {
     super(ConfigRequestType.CreateSchemaPartition);
   }
 
-  public Map<String, Map<TSeriesPartitionSlot, TRegionReplicaSet>> getAssignedSchemaPartition() {
+  public Map<String, SchemaPartitionTable> getAssignedSchemaPartition() {
     return assignedSchemaPartition;
   }
 
   public void setAssignedSchemaPartition(
-      Map<String, Map<TSeriesPartitionSlot, TRegionReplicaSet>> assignedSchemaPartition) {
+      Map<String, SchemaPartitionTable> assignedSchemaPartition) {
     this.assignedSchemaPartition = assignedSchemaPartition;
   }
 
   @Override
   protected void serializeImpl(DataOutputStream stream) throws IOException {
-    stream.writeInt(ConfigRequestType.CreateSchemaPartition.ordinal());
+    try {
+      TTransport transport = new TIOStreamTransport(stream);
+      TBinaryProtocol protocol = new TBinaryProtocol(transport);
+
+      stream.writeInt(ConfigRequestType.CreateSchemaPartition.ordinal());
 
-    stream.writeInt(assignedSchemaPartition.size());
-    for (Entry<String, Map<TSeriesPartitionSlot, TRegionReplicaSet>> entry :
-        assignedSchemaPartition.entrySet()) {
-      String storageGroup = entry.getKey();
-      Map<TSeriesPartitionSlot, TRegionReplicaSet> partitionSlots = entry.getValue();
-      BasicStructureSerDeUtil.write(storageGroup, stream);
-      stream.writeInt(partitionSlots.size());
-      partitionSlots.forEach(
-          (seriesPartitionSlot, regionReplicaSet) -> {
-            ThriftCommonsSerDeUtils.serializeTSeriesPartitionSlot(seriesPartitionSlot, stream);
-            ThriftCommonsSerDeUtils.serializeTRegionReplicaSet(regionReplicaSet, stream);
-          });
+      stream.writeInt(assignedSchemaPartition.size());
+      for (Map.Entry<String, SchemaPartitionTable> schemaPartitionTableEntry :
+          assignedSchemaPartition.entrySet()) {
+        BasicStructureSerDeUtil.write(schemaPartitionTableEntry.getKey(), stream);
+        schemaPartitionTableEntry.getValue().serialize(stream, protocol);
+      }
+    } catch (TException e) {
+      throw new IOException(e);
     }
   }
 
   @Override
   protected void deserializeImpl(ByteBuffer buffer) throws IOException {
     assignedSchemaPartition = new HashMap<>();
 
-    int storageGroupNum = buffer.getInt();
-    for (int i = 0; i < storageGroupNum; i++) {
+    int length = buffer.getInt();
+    for (int i = 0; i < length; i++) {
```

Comment:
maybe storageGroupNumber is better than length.

Reply:
Fixed~